### ISSUE TITLE
Filenames on import need to be sanitized/renamed (double-quote, ampersand, colon, etc.)

### ISSUE THREAD

**Issue by leotulipan** [akosbalasko/yarle#413]
I guess this is related to #240 

In an attachment I have a double-quote character. The filename gets created under Win10 (with the official yarle binary) as **Leadership_Is_a_Contact_Sport_-_The_“Follow-up_Fac.pdf** and in the created markdown file it gets converted to **Leadership_Is_a_Contact_Sport_-_The_%E2%80%9CFollow-up_Fac.pdf** so these 3? characters encode the quote: %E2%80%9C and of course it means I cannot open the file.

I zipped up the evernote enex export as well as the yarle log and the resulting export files if someone want's to take a look.
[yarle-test.zip](https://github.com/akosbalasko/yarle/files/9830894/yarle-test.zip)

For this test I simply checked every box I could in the yarle settings to get this result.

I am not a typescript coder, but the solution from [here](https://stackoverflow.com/questions/8485027/javascript-url-safe-filename-safe-string) looks promising

```
var s = "Filename with weird characters from the enex export file";
var filename = s.replace(/[^a-z0-9]/gi, '_').toLowerCase();

```

1. This should be a config toggle "Sanitize Filenames on import"
2. It would still need to check, if a file with that name already exists and in that case add a running number to the filename, as the replacement could result in two different files/filenames to suddenly match (think: "Example File.pdf" and "example_file.pdf")

**Comment by diamondsw** [comment]
Sanitizing filenames is kind of critical; otherwise dataloss will occur. This needs to be addressed (getting close on two years now...).

**Comment by akosbalasko** [comment]
@diamondsw sanitization was implemented long ago, this issue has a different cause. 
@leotulipan Thank you for raising up this issue, I checked your configuration and found that this property: ```urlEncodeFileNamesAndLinks``` is set to true. Which exactly did what we saw here, it urlEncoded the filenames in the notes. If I set it to false, it works as expected (as well :) ).
Let me close this issue with no fix, I think the "problem" is related to the misconfiguration. However I consider to separate the two use cases and create two options, one for urlEncodeFileNames (this is what is pointed out in this issue), and urlEncodeLinks, just for the clearness.
